madmom/ml/nn/layers.py

```diff
@@ -945,3 +945,87 @@ def activate(self, data, **kwargs):
         return self.activation_fn(
             (data - self.mean) * (self.gamma * self.inv_std) + self.beta
         )
+
+
+class FunctionLayer(Layer):
+    """
+    Generic layer that applies a given function to the data.
+
+    Parameters
+    ----------
+    func : callable
+        Function to be applied to the data.
+    *args : positional arguments
+        Positional arguments to be passed to `func`.
+    **kwargs : keyword arguments
+        Keyword arguments to be passed to `func`.
+
+    """
+
+    def __init__(self, func, *args, **kwargs):
+        self.func = func
+        self.args = args
+        self.kwargs = kwargs
+
+    def activate(self, data, **kwargs):
+        """
+        Activate the layer.
+
+        Parameters
+        ----------
+        data : numpy array
+            Activate with this data.
+
+        Returns
+        -------
+        numpy array
+            Activations for this data.
+
+        """
+        return self.func(data, *self.args, **self.kwargs)
+
+
+class PadLayer(Layer):
+    """
+    Padding layer that pads the input with a constant value.
+ 
+    Parameters
+    ----------
+    width : int
+        Width of the padding (only one value for all dimensions)
+    axes : iterable
+        Indices of axes to be padded
+    value : float
+        Value to be used for padding.
+
+    """
+
+    def __init__(self, width, axes, value=0.):
+        self.width = width
+        self.axes = axes
+        self.value = value
```

Comment:
Maybe we should allow literals like 'reflect', 'repeat' or add a separate `mode` like [SciPy does for convolution](https://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.ndimage.filters.convolve.html) in order to be more flexible in the way how to pad the data (see below).

Reply:
That would be nice, but I'd rather implement this when it is needed. Judging from the current literature, everyone uses 0-padding in NN, because it does not make a difference.